app/controllers/mixins/generic_button_mixin.rb

```diff
@@ -102,5 +102,25 @@ def evaluate_button(action, allowed_actions)
       send_action = allowed_actions[action]
       send(send_action)
     end
+
+    def validate_item_supports_action_button(action, item_type)
+      unless checked_item_id
+        error_msg = (_("\"%{action}\" requires a single item to be selected.") % {
+          :action => SupportsFeatureMixin::QUERYABLE_FEATURES[action]
+        })
+        return {:action_supported => false, :message => error_msg}
+      end
+
+      item = find_record_with_rbac(item_type, checked_item_id)
+      if item.supports?(action)
+        return {:action_supported => true, :message => nil}
```

Comment:
```
return {:action_supported => true, :message => nil} if item.supports?(action)
```
and then you don't need to indent the block below

Reply:
done